### DISCUSSION TITLE
Some packages show - ERROR: No README data found!

### DISCUSSION THREAD

**Discussion by sreejaptfa** [verdaccio/verdaccio#3105] [Q&A]
Hi @juanpicado , 
We upgraded our Verdaccio instance to 5.8.0 version. Some packages on our https://npm.<company>.org/ display the error:

ERROR: No README data found!
![verdaccio-readme](https://user-images.githubusercontent.com/9541222/161322103-b3718168-b9be-4ecc-abdd-588a5ebf750e.jpg)

Please provide your input on how to resolve this issue.

Thank you!

**Comment by juanpicado** [comment]
That message is shown when there is no readme in the package. So you might need to review your publish library process cc: #1905 . 

I have to say nothing meaningful has changed in the last release regarding that feature. 

https://github.com/verdaccio/verdaccio/issues?q=is%3Aissue+label%3A%22topic%3A+readme%22+is%3Aclosed

**Reply by jaysonpotter** [reply]
Hello @juanpicado ! 

What is interesting is that we do have readme.md files in all of our packages. Some are showing up and some aren't. I downloaded the packages from Verdaccio to make sure the readme files were really there and sure nuff, readme.md was in the package that was not displaying the and in the package that did display the readme. So not sure why some are displaying and some are not.

**Reply by edclement** [reply]
I would ask if there is any difference in capitalization? I just did a quick search of the Verdaccio codebase (v6) and most (maybe all) places in the `.ts` file where `README.md` is referenced it seems to expect all caps. You specifically state _readme.md_ (lowercase) in your response.

**Reply by juanpicado** [reply]
It does not make a difference as far I know, the package managers read this file (except yarn 2+ #1905 ) and include the content as a string in the payload on publish, so verdaccio captures this data and includes it into the storage package.json file eg: https://registry.verdaccio.org/foo

Verdaccio only keeps the latest publish readme for the `latest` tag (thinking about how to make it configurable in the future), if the readme is not in the body it returns a default empty message as the image shown.

**Reply by jaysonpotter** [reply]
@edclement I suspected the same thing till I saw the readme showing up for other packages while named "readme.md". Thank you for verifying that shouldn't make a difference @juanpicado. We will take a look into the storage package.json file to see what's going on in there. Thank you!

**Reply by juanpicado** [reply]
Last resource is debug but I am actually refactoring that part for v6 🙂 and is pretty fresh on my mind.

**Reply by jaysonpotter** [reply]
Will do! Thank you for your help. If we find our solution we will let y'all know!

**Reply by jaysonpotter** [reply]
Hey @juanpicado, We had done some shuffling of data and such in our environment that I believed caused the issue. We are restored and things are looking much better! Thank you for your help understanding how Verdaccio handles this! "The more you know" 🌈🌟

**Reply by joergrech** [reply]
Hi, are you sure that this is resolved? I just had a similar experience testing the "security" of package uploads. I published react `v18.1.3` (which does not yet exist in the npmjs registry) by increasing the version number in the JSON of the `v18.1.0` version (https://registry.npmjs.com/react/18.1.0). 
When I then unpublished the version verdaccio created/downloaded the package with the `next` tag which is currently `v18.2.0-next-e531a4a62-20220505` without any README.md (but the `next`tagged version should have a README.md: https://www.npmjs.com/package/react/v/18.2.0-next-e531a4a62-20220505).

So potentially I see two problems:
1. The README.md should have been downloaded from npmjs.com
2. I expected that verdaccio would either remove the package completely or that it would download the `latest` tag (not the `next` tag)

**Reply by juanpicado** [reply]
Ok seems a different issue, @joergrech could you create a separate discussion with the reproducible steps in detail narrow down as much you can (package managers, exact version, etc etc ...) ? Currently I'm working https://github.com/verdaccio/verdaccio/pull/3100 on refactoring the whole publish process (I've already found some issues) so it would be a good moment to test my PR with your steps and see whether I can reproduce and fix it right there.
